### ISSUE TITLE
is Priority the order of the jobs?

### ISSUE THREAD

**Issue by FrEaKmAn** [yigit/android-priority-jobqueue#28]
Hi

I'm using your great library to create jobs to sync with our API. I can easily define jobs and run job manager (jobManager.start()).

I'm syncing gallery and it's images with description. So I first add a GalleryJob with priority 1, then GalleryImageJob with priority 2. GalleryImageJob needs to access Gallery from the database, so GalleryJob has to run before GalleryImageJob.

But even if I defined priorities, it seems that sometimes GalleryImageJob (with lower priority = 2) is ran before GalleryJob (priority = 1). Is this normal?

**Comment by yigit** [comment]
Priority means order when resources are scarce. So lets say if there is 1 thread available and many jobs waiting. In that case, JobManager will select the job with the highest priority among the jobs that are ready to be run. One caveat is that, non-persistent jobs are always prioritized against  persistent jobs.

In your use case, assuming you first add GalleryJob then GalleryImageJob; it probably starts both of them because there are two available worker threads to run them.

You can ensure order by setting the same group id to both jobs. Group id means only 1 job with the same group id will be running at a given time. 
In your use case, If GalleryImageJob is something you do not want to run in parallel, just use the same group id for both GalleryJob and GalleryImageJob. If GalleryImageJob have multiple instances (e.g. 1 per image?), then add those after GalleryJob succeeds.

**Comment by yigit** [comment]
closing for inactivity.

**Comment by NightSkyDev** [comment]
@yigit - thanks for providing this answer.  It's helpful!
